### PR TITLE
Add a flag to minikube start to allow passing environment variables t…

### PR DESCRIPTION
**Pull request by dlorenc**
…o Docker.

### CODE REVIEW

**Comment by dlorenc** [comment]
This will help with #336.

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -159,5 +161,7 @@ func init() {
 	startCmd.Flags().IntVarP(&cpus, "cpus", "", constants.DefaultCPUS, "Number of CPUs allocated to the minikube VM")
 	diskFlag := startCmd.Flags().VarPF(disk, "disk-size", "", "Disk size allocated to the minikube VM (format: <number>[<unit>], where unit = b, k, m or g)")
 	diskFlag.DefValue = constants.DefaultDiskSize
+
+	startCmd.Flags().StringSliceVarP(&dockerEnv, "docker-env", "", []string{}, "Environment variables to pass to the Docker daemon. (format: key=value)")
```
Comment (jimmidyson) [inline cmd/minikube/cmd/start.go]: if there's no shorthand fort the flag you can use `StringSliceVar` &  drop the empty string arg.

Also, is it more idiomatic to pass nil for an empty slice?

Comment (dlorenc) [inline cmd/minikube/cmd/start.go]: > if there's no shorthand fort the flag you can use StringSliceVar & drop the empty string arg.

Nice.

> Also, is it more idiomatic to pass nil for an empty slice?

Switched it. I never remember that nil slices are ok...

File: pkg/minikube/cluster/cluster.go [hunk 2]
```diff
@@ -282,17 +283,29 @@ func SetupCerts(d drivers.Driver) error {
 	return nil
 }
 
+func engineOptions(config MachineConfig) *engine.Options {
+
+	o := engine.Options{
+		Env: config.DockerEnv,
```
Comment (jimmidyson) [inline pkg/minikube/cluster/cluster.go]: So this expects to be a slice of strings in the format `name=value`?

Comment (dlorenc) [inline pkg/minikube/cluster/cluster.go]: Yeah, added a comment to the struct to clarify as well. libmachine handles the parsing.

**Comment by dlorenc** [comment]
@minikube-bot retest this please

**Comment by jimmidyson** [comment]
LGTM

**Pull request merged**